root/static/scripts/edit/URLCleanup.js

```diff
@@ -1747,6 +1747,28 @@ const CLEANUPS = {
       return /^https:\/\/myspace\.com\//.test(url);
     },
   },
+  'napster': {
+    match: [new RegExp('^(https?://)?([a-z]{2}\\.)?napster\\.com', 'i')],
+    type: LINK_TYPES.streamingpaid,
+    clean: function (url) {
+      url = url.replace(/^http:\/\//, 'https://');
+      // Standardise on US (host country) for multi-country redirect 
+      url = url.replace(/^https:\/\/napster/, 'https://us.napster');
+      url = url.replace(/[#?].*$/, '');
+      return url;
+    },
+    validate: function (url, id) {
+      switch (id) {
+        case LINK_TYPES.streamingpaid.artist:
+          return /^https:\/\/[a-z]{2}\.napster\.com\/artist\/[^\/]+$/.test(url);
```

Comment:
`[^\/]+` still matches `?` and `#` too, `[\w-]+` doesn’t.

Reply:
I didn't think it'd matter, they should probably still be valid? But since they should be cleaned up before, I guess I might as well make the change :)

Reply:
Thanks, it also checks URLs have been cleaned as expected when there is some normalization.